flink-connectors/flink-sql-connector-kafka-0.10/pom.xml

```diff
@@ -41,6 +41,14 @@ under the License.
 			<artifactId>flink-connector-kafka-0.10_${scala.binary.version}</artifactId>
 			<version>${project.version}</version>
 		</dependency>
+		<!--  Due to the scope of `provided` transitive dependencies problem,
```

Comment:
this looks hacky; why can't we search the pom for dependencies with a scala-suffix instead? Shouldn't this be as simple as adding `*:*_2.1*:*` as an artifact pattern to `-Dincludes`?

Reply:
I think current PR is a more conservative approach. Do you mean that if a flink module depends on another flink module with a scala suffix, it is also considered dependent on scala? I considered this rule, but I am not sure if it was correct. Such as a module which depends on a scala-suffix module with "provided" scope, and only uses the interfaces provided by the scala-suffix module itself, regardless of the actual scala version of the runtime. May such a module exist?

Reply:
> Do you mean that if a flink module depends on another flink module with a scala suffix, it is also considered dependent on scala?

Yes. It doesn't matter whether it is provided or not.
The only exception are dependencies with a `test` scope.

Reply:
Thanks @WeiZhong94 @zentol, for @zentol's suggestion also make sense to me, just like we discuss offline @WeiZhong94 . and I mentioned above, we can take `approach 1` for this PR. 
I'll update the PR soon.